src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -91,4 +93,83 @@ private void sortSections() {
             return 1;
         });
     }
+
+    public void removeStation(Station station) {
+        validateStationToRemove(station);
+        if (isEndStation(station)) {
+            removeEndStation(station);
+            return;
+        }
+        removeMiddleStation(station);
+    }
+
+    private void validateStationToRemove(Station station) {
+        validateSectionMinimumLength();
+        validateStationExists(station);
+    }
+
+    private void validateSectionMinimumLength() {
+        if (sections.size() <= MINIMUM_SECTION_LENGTH) {
+            throw new IllegalArgumentException(
+                    String.format(ErrorMessage.ERROR_CANNOT_DELETE_SECTION_MINIMUM_LENGTH, MINIMUM_SECTION_LENGTH)
+            );
+        }
+    }
+
+    private void validateStationExists(Station station) {
+        if (!hasStation(station)) {
+            throw new IllegalArgumentException(
+                    String.format(ErrorMessage.ERROR_CANNOT_DELETE_SECTION_NOT_EXIST, station.getName())
+            );
+        }
+    }
+
+    private void removeEndStation(Station station) {
+        sections.remove(getEndSection(station));
+    }
+
+    private void removeMiddleStation(Station station) {
+        Section previousSection = getPreviousSection(station);
+        Section postSection = getPostSection(station);
+        previousSection.mergeWith(postSection);
+        sections.remove(postSection);
+    }
+
+    private Section getPreviousSection(Station station) {
+        return sections.stream()
+                .filter(section -> section.getDownStation().equals(station))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException(
+                        String.format(ErrorMessage.ERROR_PREVIOUS_SECTION_NOT_FOUND, station.getName()))
+                );
+    }
+
+    private Section getPostSection(Station station) {
+        return sections.stream()
+                .filter(section -> section.getUpStation().equals(station))
```

Comment:
```suggestion
                .filter(section -> section.sameUpStation(station))
```
객체에게 `equals` 검증을 시켜본다면 어떨까요?

Reply:
훨씬 깔끔한 것 같아요! 😃

Reply:
이미 아실수도 있지만, 한번더 상기시켜드려요! [디미터 법칙](https://tecoble.techcourse.co.kr/post/2020-06-02-law-of-demeter/)